src/invoke_training/training/finetune_lora/finetune_lora_sdxl.py

```diff
@@ -225,6 +220,43 @@ def _encode_prompt(text_encoders: list[CLIPPreTrainedModel], prompt_token_ids_li
     return prompt_embeds, pooled_prompt_embeds
 
 
+def _cache_text_encoder_outputs(
+    cache_dir: str,
+    config: FinetuneLoRASDXLConfig,
+    tokenizer_1: PreTrainedTokenizer,
+    tokenizer_2: PreTrainedTokenizer,
+    text_encoder_1: CLIPPreTrainedModel,
+    text_encoder_2: CLIPPreTrainedModel,
+):
+    """Run the text encoder on all captions in the dataset and cache the results to disk.
+    Args:
+        cache_dir (str): The directory where the results will be cached.
+        config (FinetuneLoRAConfig): Training config.
+        tokenizer_1 (PreTrainedTokenizer):
+        tokenizer_2 (PreTrainedTokenizer):
+        text_encoder_1 (CLIPPreTrainedModel):
+        text_encoder_2 (CLIPPreTrainedModel):
+    """
+    data_loader = build_image_caption_sdxl_dataloader(
+        config.dataset, tokenizer_1, tokenizer_2, config.train_batch_size, shuffle=False
+    )
+
+    cache = TensorDiskCache(cache_dir)
```

Comment:
how big can this get?

Reply:
On a test dataset with 833 images, it is ~250MB. So, large enough that it is probably worth cleaning up. I'll add some cleanup logic.

Reply:
Switched to using a temporary directory in [bf220f9](https://github.com/invoke-ai/invoke-training/pull/17/commits/bf220f9f4a91d343ed41024fdf2d53f44788bc87)